app/src/main/java/org/fossasia/openevent/app/data/AbstractObservable.java

```diff
@@ -18,15 +19,19 @@
 public final class AbstractObservable {
 
     private final ConnectionStatus connectionStatus;
+    private final RateLimiter<String> rateLimiter = new RateLimiter<>(Duration.ofMinutes(10));
 
     public static final class AbstractObservableBuilder<T> {
+        private final AbstractObservable abstractObservable;
         private final ConnectionStatus connectionStatus;
         private boolean reload;
+        private String rateLimiterKey;
         private Observable<T> diskObservable;
         private Observable<T> networkObservable;
 
-        public AbstractObservableBuilder(ConnectionStatus connectionStatus) {
+        public AbstractObservableBuilder(ConnectionStatus connectionStatus, AbstractObservable abstractObservable) {
```

Comment:
Since we are building abstract observable, I don't think it'd be good to ask it in constructor